README.md

```diff
@@ -1,143 +1,431 @@
 
-A Julia package to play and analyze [Wordle](https://en.wikipedia.org/wiki/Wordle) and related games, such as [Primel](https://cojofra.github.io/primel/).
+[![Stable](https://img.shields.io/badge/docs-stable-blue.svg)](https://dmbates.github.io/Wordlegames.jl/stable)
+[![Dev](https://img.shields.io/badge/docs-dev-blue.svg)](https://dmbates.github.io/Wordlegames.jl/dev)
+[![Build Status](https://github.com/dmbates/Wordlegames.jl/actions/workflows/CI.yml/badge.svg?branch=main)](https://github.com/dmbates/Wordlegames.jl/actions/workflows/CI.yml?query=branch%3Amain)
+[![Coverage](https://codecov.io/gh/dmbates/Wordlegames.jl/branch/main/graph/badge.svg)](https://codecov.io/gh/dmbates/Wordlegames.jl)
+[![Code Style: Blue](https://img.shields.io/badge/code%20style-blue-4495d1.svg)](https://github.com/invenia/BlueStyle)
+[![PkgEval](https://JuliaCI.github.io/NanosoldierReports/pkgeval_badges/E/Wordlegames.svg)](https://JuliaCI.github.io/NanosoldierReports/pkgeval_badges/report.html)
 
-A game is represented by a `GamePool` constructed from the list of possible targets
+This [Julia](https://julialang.org) package allows for playing and analyzing [Wordle](https://en.wikipedia.org/wiki/Wordle) and related games, such as [Primel](https://cojofra.github.io/primel/).
+
+A game is represented by a `GamePool` of targets, potential guesses, and some game play status information.
+By default the game is played as in the "Hard Mode" setting on the Wordle app and web site, which means that the only guesses allowed at each turn are those in the current target pool.
+As a consequence, the initial pool of potential guesses is the same as the initial target pool.
+
+Consider a game played on the [Primel site](https://cojofra.github.io/primel/) on 2022-02-11 resulting in the score
+
+```
+Primel 237 4/6*
+
+⬜🟨⬜⬜🟨
+⬜🟨🟨🟨🟩
+🟨🟨🟨⬜🟩
+🟩🟩🟩🟩🟩
+```
+
+The goal of the game is to guess a 5-digit prime number (i.e. between 10,000 and 99,999) where each guess is scored as a pattern of 5 tiles colored gray, yellow, or green.
+
+Here the scores are entered as 0, 1, or 2 instead of gray, yellow, or green.
 
 ```jl
-julia> using Wordlegames
+julia> using Primes, Wordlegames
+
+julia> primel = GamePool(primes(10000, 99999));
+
+julia> (string(last(primel.guesses)...), last(primel.poolsizes), last(primel.expected), last(primel.entropy))
+("17923", 8363, 121.54167f0, 6.62459f0)
+
+julia> scoreupdate!(primel, [0,1,0,0,1]);
+
+julia> (string(last(primel.guesses)...), last(primel.poolsizes), last(primel.expected), last(primel.entropy))
+("56437", 206, 9.495146f0, 4.76909f0)
+
+julia> scoreupdate!(primel, [0,1,1,1,2]);
+
+julia> (string(last(primel.guesses)...), last(primel.poolsizes), last(primel.expected), last(primel.entropy))
+("34607", 10, 1.4f0, 2.921928f0)
+
+julia> scoreupdate!(primel, [1,1,1,0,2]);
+
+julia> (string(last(primel.guesses)...), last(primel.poolsizes), last(primel.expected), last(primel.entropy))
+("43867", 2, 1.0f0, 1.0f0)
```

Comment:
I found this really hard to follow -- could you add some comments? For example, since I know the congruence class trick, I understand why the score update works, but that's not clear from the text so far.